pool.go

```diff
@@ -339,45 +339,43 @@ func (p *Pool) retrieveWorker() (w worker) {
 	}
 
 	p.lock.Lock()
-	w = p.workers.detach()
-	if w != nil { // first try to fetch the worker from the queue
+
+retry:
+	// First try to fetch the worker from the queue.
+	if w = p.workers.detach(); w != nil {
 		p.lock.Unlock()
-	} else if capacity := p.Cap(); capacity == -1 || capacity > p.Running() {
-		// if the worker queue is empty and we don't run out of the pool capacity,
-		// then just spawn a new worker goroutine.
+		return
+	}
+
+	// If the worker queue is empty and we don't run out of the pool capacity,
+	// then just spawn a new worker goroutine.
+	if capacity := p.Cap(); capacity == -1 || capacity > p.Running() {
 		p.lock.Unlock()
 		spawnWorker()
-	} else { // otherwise, we'll have to keep them blocked and wait for at least one worker to be put back into pool.
-		if p.options.Nonblocking {
-			p.lock.Unlock()
-			return
-		}
-	retry:
-		if p.options.MaxBlockingTasks != 0 && p.Waiting() >= p.options.MaxBlockingTasks {
-			p.lock.Unlock()
-			return
-		}
+		return
+	}
+
+	// Otherwise, we'll have to keep them blocked and wait for at least one worker to be put back into pool.
+	if p.options.Nonblocking {
```

Comment:
I think we can combine this `if` condition with the following one.

Reply:
Got it!

Reply:
We can add a new comment on it after we combine the two below `if` blocks, something like:
```
// Bail out early if it's in nonblocking mode or the number of pending callers reaches the maximum limit value.
```

Reply:
You meant put the comment below or above?
```go
	// Above
	// Bail out early if it's in nonblocking mode or the number of pending callers reaches the maximum limit value.
	if p.options.Nonblocking || (p.options.MaxBlockingTasks != 0 && p.Waiting() >= p.options.MaxBlockingTasks) {
		// Below
		// Bail out early if it's in nonblocking mode or the number of pending callers reaches the maximum limit value.
		p.lock.Unlock()
		return
	}
```

Reply:
Put it above.